components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -971,10 +978,14 @@ subroutine p3_main_part3(kts, kte, kbot, ktop, kdir, &
 
       ! rain:
       if (qr(k).ge.qsmall) then
-
-         call get_rain_dsd2(qr(k),nr(k),mu_r(k),lamr(k),tmp1,tmp2,cld_frac_r(k))
-
-         ze_rain(k) = nr(k)*(mu_r(k)+6._rtype)*(mu_r(k)+5._rtype)*(mu_r(k)+4._rtype)*           &
+         qr_incld = qr(k)/cld_frac_r(k)
+         nr_incld = nr(k)/cld_frac_r(k)
+         call get_rain_dsd2(qr_incld,nr_incld,mu_r(k),lamr(k),tmp1,tmp2,cld_frac_r(k))
+         nr(k) = nr_incld*cld_frac_r(k) !limiters might change nc_incld... enforcing consistency
+
+         !PMC potential bug: changed nr to nr_incld below. now ze is an in-cloud ze, while users
+         !may be assuming ze is a cell-averaged thing.
+         ze_rain(k) = nr_incld*(mu_r(k)+6._rtype)*(mu_r(k)+5._rtype)*(mu_r(k)+4._rtype)*           &
```

Comment:
What would users expect for ze (radar reflectivity)? Before this PR, stuff was a mishmash of in-cloud and cell-ave quantities. Now ze_rain is computed using all in-cloud values. Would people looking at output expect ze_rain to be cell ave? If so, can I just multiply it by the cloud fraction, or do I need to warp the drop size distribution in some way. @brhillman - do you have an opinion? @JS-WRF-SBM ? @crterai ? @hassanbeydoun ?

Reply:
Since I think ze_rain is just a diagnostic output, the easiest thing would be to just delete ze_rain (in which case this get_rain_dsd2 call may not be needed at all? Then whoever wants ze can implement what they want.

Reply:
I'm not following why after sedimentation where we update cell-avg values, you decided to go back to in-cld values here at the part3 routine.
The tables doesn't 'understand' what value are in the input, we just need to maintain consistency (and we are, before sedimentation)

Regarding Ze calc, I think that cell-avg is more coherent with what is actually being observed (volumetric scanning values), especially for high-res. But for low-res it might be less representable.
Also, if the riming and Ze_ice is cell-avg, Ze_rain should be the same (or the opposite way around).

Reply:
update regarding ze: I think the original implementation was probably correct but terribly confusing. It only makes sense to compute things involving parameters of our rain size distribution using in-rainy-area values because the rain size PDF doesn't exist/make sense outside of the cloudy region... so nr in the line above really should be nr_incld (like I changed it to be). But then we should convert the result to a cell ave value via ze_cell_ave = cld_frac_r*ze_incld + (1-cld_frac_r)*0. This ends up being ze = cld_frac_r*nc/cld_frac_r * (stuff above)... which is equal to what's written above. Seems like we need to add a clarifying comment here!!!